### PR TITLE
Ensure RUBY_ALIGNOF == the "alignment requirement" defined in C11

### PR DESCRIPTION
**Pull request by shyouhei**
Context: https://github.com/ruby/ruby/pull/3261#discussion_r491698137

GCC/clang have `__alignof__` operator.  This is very much like C11 `_Alignof`, but can be different on a i386 System V ABI situation.  The difference is troublesome.  We would like to stick to what C11 says.

This is how we achieve that.

### CODE REVIEW

**Comment by shyouhei** [comment]
Am checking icc / sunc...

**Comment by shyouhei** [comment]
Both icc/sunc seems working _Alignof, no problem observed for me.

**Comment by akr** [comment]
It seems fine.

I recommend some comment about a struct with flexible array member, though.

**Review by mrkn — DISMISSED** [review]

**Review by mrkn — COMMENTED** [review]
Is it enough to check only `double`?

**Comment by shyouhei** [comment]
@mrkn I think so.  There are other types that exhibit similar behaviour (e.g. `unsigned long long`), but that doesn't necessarily mean we should also check them.  If `double` is NG everything else shall also be NG.

**Review by mrkn — APPROVED** [review]

File: include/ruby/internal/stdalign.h [hunk 1]
```diff
@@ -75,50 +73,61 @@
...
-# define RBIMPL_ALIGNOF __alignof__
-
-#elif 0
-# /* THIS IS NG, you cannot define a new type inside of offsetof. */
+#else
+# /* :BEWARE:  This is  the last  resort.   If your  compiler somehow  supports
+#  * querying the alignment of a type,  you definitely should use that instead.
+#  * There are 2 known pitfalls for this fallback implementation:
+#  *
+#  * Fitst, it is either an undefined  behaviour (C) or an explicit error (C++)
```
Comment (Maumagnaguagno) [inline include/ruby/internal/stdalign.h:119]: Fitst => First

**Comment by pNm193** [comment]
Alignof (T, t) issue?

**Pull request merged**